src/helpers/sso-helper.ts

```diff
@@ -3,34 +3,40 @@
  * See LICENSE in the project root for license information.
  */
 
-import { dialogFallback } from "./fallbackauthhelper";
-import { handleClientSideErrors, makeGraphApiCall, showMessage } from "office-addin-sso";
+import { dialogFallback } from "./fallbackauthdialog";
+import { callGetUserData } from "./middle-tier-calls";
+import { showMessage } from "./message-helper";
+import { handleClientSideErrors } from "./error-handler";
 
 /* global OfficeRuntime */
 
-let retryGetAccessToken = 0;
+let retryGetMiddletierToken = 0;
 
-export async function getGraphData(callback): Promise<void> {
+export async function getUserData(callback): Promise<void> {
   try {
-    let bootstrapToken: string = await OfficeRuntime.auth.getAccessToken({ allowSignInPrompt: true });
-    let response: any = await makeGraphApiCall(bootstrapToken);
-    if (response.claims) {
+    let middletierToken: string = await OfficeRuntime.auth.getAccessToken({
+      allowSignInPrompt: true,
+      allowConsentPrompt: true,
+      forMSGraphAccess: true,
+    });
+    let response: any = await callGetUserData(middletierToken);
+    if (!response) {
+      return Promise.reject();
```

Comment:
Kind of confusing to create an async function, but then use Promises. Why not throw an error here just stay consistent with the async/await pattern.

Reply:
From what I understand an async function is a promise, it's just syntax difference.  That said, I think I understand what you're suggesting as far practice goes for mixing the syntax.  Will update to be more consistent.